Usage of gpt-3.5-turbo model

Hello,

I was trying to see if I can modify the model from text-davinci-003 to gpt-3.5-turbo in vector search but noticed that the app doesn't work anymore and keep getting search errors. 
Is it any error from my end or some other issue that I am unaware of

Hello @gregnr I tried to replicate the code shared by you but I keep getting the error
Type error: Cannot find module 'import { serve } from 'https://deno.land/std@0.170.0/http/server.ts'' or its corresponding type declarations.

> 1 | import { serve } from 'https://deno.land/std@0.170.0/http/server.ts'';
    |                       ^
  2 | import 'xhr@0.2.1/mod.ts'
  3 | import { createClient } from '@supabase/supabase-js'
  4 | import { codeBlock, oneLine } from 'common-tags'
 ELIFECYCLE  Command failed with exit code 1.
 Any idea on how to resolve as I managed to install local packages for others but unable for Deno

I am getting the error when deploying it on vercel.

Including the error message I got after making another attempt today




Attention: Next.js now collects completely anonymous telemetry regarding usage.
--
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry
 
- info Linting and checking validity of types...
Failed to compile.
 
./pages/api/vector-search.ts:1:23
Type error: Cannot find module 'https://deno.land/std@0.186.0/http/server.ts' or its corresponding type declarations.
 
> 1 \| import { serve } from "https://deno.land/std@0.186.0/http/server.ts";
\|                       ^
2 \| import 'x/xhr@0.2.1/mod.ts'
3 \| import { createClient } from '@supabase/supabase-js'
4 \| import { codeBlock, oneLine } from 'common-tags'
ELIFECYCLE  Command failed with exit code 1.
Error: Command "pnpm run build" exited with 1
Deployment completed
BUILD_UTILS_SPAWN_1: Command "pnpm run build" exited with 1

--- a/pages/api/vector-search.ts
+++ b/pages/api/vector-search.ts
@@ -7,6 +7,7 @@ import {
   OpenAIApi,
   CreateModerationResponse,
   CreateEmbeddingResponse,
+  ChatCompletionRequestMessage,
 } from 'openai-edge'
 import { OpenAIStream, StreamingTextResponse } from 'ai'
 import { ApplicationError, UserError } from '@/lib/errors'
@@ -130,9 +131,14 @@ export default async function handler(req: NextRequest) {
       Answer as markdown (including related code snippets if available):
     `
 
-    const response = await openai.createCompletion({
-      model: 'text-davinci-003',
-      prompt,
+    const chatMessage: ChatCompletionRequestMessage = {
+      role: 'user',
+      content: prompt,
+    }
+
+    const response = await openai.createChatCompletion({
+      model: 'gpt-3.5-turbo',
+      messages: [chatMessage],
       max_tokens: 512,
       temperature: 0,
       stream: true,

